### PR TITLE
Implemented sonar pulse, fixed some style issues.

### PR DESCRIPTION
**Pull request by Crollt**

### CODE REVIEW

**Comment by Flazzing** [comment]
Great job

File: src/main/java/assets/css/custom.css [hunk 1]
```diff
@@ -40,6 +39,7 @@ header{
     margin-right: auto;
     align-items: center;
     width: 93%;
+    opacity:0.8;
```
Comment (Bizjakj) [inline src/main/java/assets/css/custom.css]: good looking out on the header

File: src/main/java/controllers/SonarGameAction.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package controllers;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import cs361.battleships.models.Game;
+
+public class SonarGameAction {
+
+    @JsonProperty
+    private Game game;
```
Comment (Bizjakj) [inline src/main/java/controllers/SonarGameAction.java]: does the game variable on the sonar just hold the tiles that have been sonared?

File: src/main/java/cs361/battleships/models/Board.java [hunk 3]
```diff
@@ -142,6 +146,40 @@ public Result attack(int x, char y) {
...
+	    // Add circle pattern to squares
+	    for (int i = 0; i < radius+1; i++) {
+			squares.add(new Square(x+i, y));
+			squares.add(new Square(x-i, y));
+			squares.add(new Square(x, (char)((int)y+i)));
+			squares.add(new Square(x, (char)((int)y-i)));
+			squares.add(new Square(x-1, (char)((int)y-1)));
+			squares.add(new Square(x-1, (char)((int)y+1)));
+			squares.add(new Square(x+1, (char)((int)y-1)));
+			squares.add(new Square(x+1, (char)((int)y+1)));
```
Comment (Bizjakj) [inline src/main/java/cs361/battleships/models/Board.java]: is the type casting neeeded here?

File: src/main/java/cs361/battleships/models/Game.java [hunk 4]
```diff
@@ -49,6 +49,13 @@ public boolean attack(int x, char y) {
         return true;
     }
 
+    public boolean sonar(int x, char y, int numSonars) {
+        boolean successful = opponentsBoard.sonar(x, y, numSonars);
+        if (!successful)
+            return false;
+        return true;
+    }
```
Comment (Bizjakj) [inline src/main/java/cs361/battleships/models/Game.java]: seems a bit more clear to just return opponentsBoard.sonar(x, y, numSonars); right away, or just successful

Comment (Bizjakj) [reply src/main/java/cs361/battleships/models/Game.java]: also does opponentsBoard.sonar call a different function? if it calls this function which calls itself recursively what is it doing.

File: src/main/java/assets/css/custom.css (hunk 1 above)
Comment (Crollt) [reply src/main/java/assets/css/custom.css]: ;)

File: src/main/java/cs361/battleships/models/Board.java (hunk 3 above)
Comment (Crollt) [reply src/main/java/cs361/battleships/models/Board.java]: Not sure, it works though. I think it's a little more clear as well.

File: src/main/java/cs361/battleships/models/Game.java (hunk 4 above)
Comment (Crollt) [reply src/main/java/cs361/battleships/models/Game.java]: No this is better...

opponentsBoard.sonar is in the Board class. It handles the sonar.

**Pull request merged**